aquasec/resource_application_scope.go

```diff
@@ -402,14 +405,14 @@ func expandApplicationScope(d *schema.ResourceData) (*client.ApplicationScope, e
 
 func resourceApplicationScopeRead(d *schema.ResourceData, m interface{}) error {
 	ac := m.(*client.Client)
-	name := d.Get("name").(string)
-
-	iap, err := ac.GetApplicationScope(name)
+	iap, err := ac.GetApplicationScope(d.Id())
 	if err == nil {
+
 		d.Set("name", iap.Name)
 		d.Set("description", iap.Description)
 		d.Set("author", iap.Author)
 		d.Set("owner_email", iap.OwnerEmail)
+		d.SetId(iap.Name)
```

Comment:
why to set it twice? first time is in line: 341

Reply:
341 is create function this is read?

Reply:
yes, but the only use with the read is after create, so basically you are setting it twice

Reply:
the read is used with import as well as being called after read